lib/project_types/script/layers/infrastructure/assemblyscript_dependency_manager.rb

```diff
@@ -22,19 +22,28 @@ def installed?
         end
 
         def install
+          check_node_version!
+
           output, status = @ctx.capture2e("npm", "install", "--no-audit", "--no-optional", "--loglevel error")
           raise Errors::DependencyInstallError, output unless status.success?
         end
 
         private
 
+        def check_node_version!
+          output, status = @ctx.capture2e("node", "--version")
+          raise Errors::DependencyInstallError, output unless status.success?
+
+          version = Semantic::Version.new(output[1..-1])
+          unless version >= Semantic::Version.new("12.16.0")
```

Comment:
Question: why is this hard coded to 12.16.0?  What's the version of `npm` the app developer needs to create an app project?  Is it consistent?  Does any changes in the EP/SDK toolchain ever affect this version?

Reply:
No, it's a requirement for as-pect, since it uses Node experimental features that were initially released in 12.16.